src/main/scala/symsim/ExactRL.scala

```diff
@@ -92,11 +92,12 @@ trait ExactRL[State, ObservableState, Action, Reward, Scheduler[_]]
    *
    *  @return the target state rached and an updated reward
    */
-  def evalEpoch (p: Policy) (s_t: State, r_t: Reward): Scheduler[(State, Reward)] =
+  def evalEpoch (p: Policy) (s_t: State, r_t: Reward): List[(State, Reward)] =
```

Comment:
This is good. I was thinking in exactly the same direction. I think you probably missed a take(5) somewhere on your randomized, and this is why it took all the memory computing a mean and variance - but I totally agree that we should not abuse the fact that randomized is a list - this leads to all sorts of mistakes, so I was also thinking to change this to list, although I was more inclined to go the lazy list direction, but this is totally secondary. So let's try to keep this way.

Reply:
I am however concerned that this is not the right function to change to lists. This one still produces a single Random value (because step is random, and the types should continue capturing this).